lib/stencil-push.spec.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const Code = require('code');
+const Lab = require('lab');
+const lab = exports.lab = Lab.script();
+const describe = lab.describe;
+const expect = Code.expect;
+const it = lab.it;
+const rewire = require("rewire");
+const sinon = require('sinon');
+const StencilPush = rewire('./stencil-push');
+const Wreck = require('wreck');
+const mockDb = {
+    _data: {},
+    get data() {
+        return this._data;
+    },
+    set data(data) {
+        this._data = data;
+    },
+}
+
+describe('stencil push', () => {
+    lab.beforeEach(done => {
+        mockDb.data = {};
+
+        this.wreckGetStub = sinon.stub(Wreck, 'get').callsFake(requestStub);
+        this.wreckPostStub = sinon.stub(Wreck, 'post').callsFake(requestStub);
+        this.wreckPutStub = sinon.stub(Wreck, 'put').callsFake(requestStub);
+
+        this.restoreStencilPush = StencilPush.__set__({
+            "utils.generateBundle": utilStub({
+                bundleZipPath: 'bundleZipPath',
+            }),
+            "utils.promptUserWhetherToApplyTheme": utilStub({
+                applyTheme: true,
+            }),
+            "utils.promptUserForVariation": utilStub({
+                variationId: 'bold',
+            }),
+        });
+
+        function requestStub(url, options, callback) {
+            process.nextTick(() => {
+                callback(null, { statusCode: 200 }, mockDb.data);
+            });
+        }
+
+        function utilStub(data) {
+            return (options, callback) => {
+                process.nextTick(() => {
+                    callback(null, Object.assign({}, options, data));
+                });
+            };
+        }
+
+        done();
+    });
+
+    lab.afterEach(done => {
+        this.wreckGetStub.restore();
+        this.wreckPostStub.restore();
+        this.wreckPutStub.restore();
+        this.restoreStencilPush();
```

Comment:
Can you use `sandbox` instead? that way you don't need to restore all stubs, just call `sandbox.restore()`

https://github.com/bigcommerce/stencil-cli/blob/f3fe45612ae094d5d0dccc0cffb99acdf8959bd3/bin/stencil-bundle.spec.js#L43